src/main/java/opendcs/opentsdb/OpenTimeSeriesDAO.java

```diff
@@ -830,72 +830,90 @@ public void deleteTimeSeriesRange(CTimeSeries ts, Date from, Date until)
 		{
-			warning("deleteTimeSeries error deleting alarm records: " + ex);
-		}
-		finally
-		{
-			alarmDAO.close();
-		}
-
-		CwmsTsId ctsid = (CwmsTsId)tsid;
-		String tableName = makeDataTableName(ctsid);
-		String q = "delete from " + tableName
-			+ " where ts_id = " + ctsid.getKey()
-			+ " and flags & " + CwmsFlags.PROTECTED + " = 0 ";
-		doModify(q);
-		doModify("delete from ts_property where ts_id = " + ctsid.getKey());
-		
-		CompDependsDAI compDependsDAO = db.makeCompDependsDAO();
-		try
-		{
-			compDependsDAO.deleteCompDependsForTsKey(ctsid.getKey());
-		}
-		catch(Exception ex)
-		{
-			warning("deleteTimeSeries error deleting computation dependencies: " + ex);
-		}
-		finally { compDependsDAO.close(); }
-		
-		doModify("delete from cp_comp_tasklist where ts_id = " + ctsid.getKey());
-		
-		doModify("delete from ts_spec where ts_id = " + ctsid.getKey());
-		
-		q = "select num_ts_present, est_annual_values from storage_table_list "
-			+ "where table_num = " + ctsid.getStorageTable();
-		ResultSet rs = doQuery(q);
-		try
-		{
-			if (rs != null && rs.next())
+			this.inTransaction(dao ->
 			{
-				int num = rs.getInt(1);
-				int estValues = rs.getInt(2);
-				num--;
-				estValues -= interval2estAnnualValues(ctsid.getIntervalOb());
-				q = "update storage_table_list set num_ts_present = " + num
-					+ ", est_annual_values = " + estValues
-					+ " where table_num = "+ ctsid.getStorageTable();
-				doModify(q);
-			}
-		}
-		catch(SQLException ex)
-		{
-			warning("Error in query '" + q + "': " + ex);
+				alarmDAO.inTransactionOf(dao);
+				compDependsDAO.inTransactionOf(dao);
+				try
+				{
+					alarmDAO.deleteCurrentAlarm(tsid.getKey(), null);
+					alarmDAO.deleteHistoryAlarms(tsid.getKey(), null, null);
+				}
+				catch(Exception ex)
+				{
+					throw new Exception("error deleting alarm records.", ex);
+				}
+
+				try
+				{
+					String tableName = makeDataTableName(ctsid);
+					String q = "delete from " + tableName
+						+ " where ts_id = ?"
+						+ " and flags & ? = 0 ";
+					dao.doModify(q, ctsid.getKey(), CwmsFlags.PROTECTED);
+					dao.doModify("delete from ts_property where ts_id = ?", ctsid.getKey());
+					compDependsDAO.deleteCompDependsForTsKey(ctsid.getKey());
+				}
+				catch(Exception ex)
+				{
+					throw new Exception("deleteTimeSeries error deleting computation dependencies.",ex);
+				}
+
+				try
+				{
+					dao.doModify("delete from cp_comp_tasklist where ts_id = ?", ctsid.getKey());
+				}
+				catch (Exception ex)
+				{
+					throw new Exception("Error removing tasklist entries", ex);
+				}
+
+				try
+				{
+					dao.doModify("delete from ts_spec where ts_id = ?", ctsid.getKey());
+				}
+				catch (Exception ex)
+				{
+					throw new Exception("Error deleting ts_spec entry", ex);
+				}
+
+				try
+				{
+					String q = "select num_ts_present, est_annual_values from storage_table_list "
+					+ "where table_num = ?" + ctsid.getStorageTable();
```

Comment:
```suggestion
					String q = "select num_ts_present, est_annual_values from storage_table_list "
					+ "where table_num = ?";
```
Doesn't seem quite right, but not sure if this is the proper fix.

Reply:
You're suggested change is correct, I missed actually deleting that particular string append.